python/cudf/cudf/core/frame.py

```diff
@@ -236,9 +237,24 @@ def _concat(cls, objs, axis=0, ignore_index=False, sort=False):
                     result_index_length += len(obj)
                     empty_has_index = empty_has_index or len(obj) > 0
 
-        # Get a list of the unique table column names
-        names = [name for f in objs for name in f._column_names]
-        names = OrderedDict.fromkeys(names).keys()
+        if join == "inner":
+            old_index = []
+            for obj in objs:
+                index = obj.columns
+                old_index.append(index)
+            index = old_index[0]
+            for other in old_index[1:]:
+                index = index.intersection(other)
```

Comment:
Feels to me `old_index` is not really needed here. Also should probably check if `old_index` is empty or not?
```suggestion
            index = objs[0].columns if len(objs) > 0 else pd.Index([])
            for other in objs[1:]:
                index = index.intersection(other.columns)
```

Reply:
Awesome! That should work as well. Looks a lot cleaner. I'll commit it and then run tests to make sure.